crates/ide_completion/src/render.rs

```diff
@@ -325,11 +327,20 @@ impl<'a> Render<'a> {
 fn compute_relevance(ctx: &RenderContext, ty: &Type, name: &str) -> Option<CompletionRelevance> {
     let (expected_name, expected_type) = ctx.expected_name_and_type()?;
     let mut res = CompletionRelevance::default();
-    res.exact_type_match = ty == &expected_type;
+    res.exact_type_match = ty == &expected_type
+        || autoderef_relevance(
+            ctx.db().upcast(),
+            ty,
+            expected_type.remove_ref().as_ref().unwrap_or(&expected_type),
+        );
```

Comment:
It feels weird that the caller needs to write "exactly equal or autoderefs" logic. It's better to encapsulate both into a single function so that the call site is simpler `exact_type_match = types_match(foo, bar)`
.

Reply:
Should I squash this new function and `autoderef_relevance` into one?

Reply:
I think that would be best, yeah!